File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
    
        HashMap<Integer, Integer> tangCnt = new HashMap<>();
        
        for(int i=0; i<tangerine.length; i++){
            tangCnt.put(tangerine[i], tangCnt.getOrDefault(tangerine[i], 0)+1);
        }
        
        PriorityQueue<int[]> pq = new PriorityQueue<>(new Comparator<int[]>(){
            @Override
            public int compare(int o1[], int o2[]){
                return o2[1] - o1[1];
            }
        });
        
        tangCnt.forEach((key, value) -> {
            pq.add(new int[] {key, value});
        });
        
        int cnt = 0;
        int answer = 0;
        
        while(cnt < k){
            int now[] = pq.poll();
            cnt += now[1];
            answer ++;
        }

        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    
    const map = new Map();
    
    for(let i=0; i<tangerine.length; i++){
        map.set(tangerine[i], (map.get(tangerine[i]) || 0) + 1);
    }
    
    const tangArr  = [];
    
    map.forEach((value, key) => {
        tangArr.push([key, value]);
    }) 
    
    tangArr.sort((a, b) => {
        return b[1] - a[1]
    })
    
    var answer = 0;
    var cnt = 0;
    
    while(cnt < k){
        cnt += tangArr[0][1];
        answer ++;
        tangArr.shift();
    }
    
    return answer;
}